### ISSUE TITLE
Special char in desc file will cause pacgraph to throw an error.

### ISSUE THREAD

**Issue by Blind55** [keenerd/pacgraph#1]
Hi,

one will get an error if the desc file contains a special character. E.g.

"Error reading package /var/lib/pacman/local/openmpi-1.5.3-3"

because the %PACKAGER% has a special char in his name.

Nice program though :)

**Comment by keenerd** [comment]
I am unable to produce this error.

```
pacgraph -m arch-repo -c openmpi
pacgraph -m arch-repo openmpi
pacgrapg-tk -m arch-repo openmpi
pacman -S openmpi; pacgraph
```

All of these work fine here.

What version of Python do you have installed?  This should just work with Python 3.

**Comment by keenerd** [comment]
I asked around and it could also be a locale bug.  What is your $LANG set to?  (It should be a utf8 locale.)

**Comment by Blind55** [comment]
Sorry for the delay...

$ echo $LANG  
en_US.UTF-8

I do have python3 installed.

To elaborate a bit:

$ pacgraph
Loading package info
Error reading package /var/lib/pacman/local/pkg-config-0.26-1
Error reading package /var/lib/pacman/local/attr-2.4.46-1
Error reading package /var/lib/pacman/local/mkinitcpio-busybox-1.18.4-1
Error reading package /var/lib/pacman/local/cyrus-sasl-plugins-2.1.23-5
Error reading package /var/lib/pacman/local/texlive-htmlxml-2010.21272-1
Error reading package /var/lib/pacman/local/libsndfile-1.0.24-1
Error reading package /var/lib/pacman/local/texlive-pictures-2010.22101-1
Error reading package /var/lib/pacman/local/gawk-3.1.8-2
Error reading package /var/lib/pacman/local/diffutils-3.0-2
Error reading package /var/lib/pacman/local/libtiff-3.9.5-1
Error reading package /var/lib/pacman/local/device-mapper-2.02.85-2
Error reading package /var/lib/pacman/local/heirloom-mailx-12.5-1
Error reading package /var/lib/pacman/local/less-443-2
Error reading package /var/lib/pacman/local/libdvdcss-1.2.10-2
Error reading package /var/lib/pacman/local/texlive-science-2010.22139-1
Error reading package /var/lib/pacman/local/python2-cairo-1.10.0-1
Error reading package /var/lib/pacman/local/gettext-0.18.1.1-3
Error reading package /var/lib/pacman/local/geeqie-1.0-5
Error reading package /var/lib/pacman/local/texlive-music-2010.21982-1
Error reading package /var/lib/pacman/local/hwloc-1.1.1-2
Error reading package /var/lib/pacman/local/gdbm-1.8.3-8
Error reading package /var/lib/pacman/local/gzip-1.4-2
Error reading package /var/lib/pacman/local/texlive-genericextra-2010.22096-1
Error reading package /var/lib/pacman/local/python-3.2-2
Error reading package /var/lib/pacman/local/which-2.20-4
Error reading package /var/lib/pacman/local/texlive-fontsextra-2010.22053-1
Error reading package /var/lib/pacman/local/popt-1.16-3
Error reading package /var/lib/pacman/local/texlive-plainextra-2010.21462-1
Error reading package /var/lib/pacman/local/libtirpc-0.2.1-3
Error reading package /var/lib/pacman/local/libssh2-1.2.7-2
Error reading package /var/lib/pacman/local/mkinitcpio-0.6.15-1
Error reading package /var/lib/pacman/local/rsync-3.0.8-2
Error reading package /var/lib/pacman/local/mplayer2-2.0-12
Error reading package /var/lib/pacman/local/automake-1.11.1-2
Error reading package /var/lib/pacman/local/patch-2.6.1-2
Error reading package /var/lib/pacman/local/nfs-utils-1.2.3-2
Error reading package /var/lib/pacman/local/texlive-latexextra-2010.22155-1
Error reading package /var/lib/pacman/local/texlive-core-2010.22154-1
Error reading package /var/lib/pacman/local/python2-matplotlib-1.0.1-2
Error reading package /var/lib/pacman/local/v4l-utils-0.8.4-1
Error reading package /var/lib/pacman/local/cryptsetup-1.3.1-1
Error reading package /var/lib/pacman/local/krb5-1.9.1-1
Error reading package /var/lib/pacman/local/ppp-2.4.5-2
Error reading package /var/lib/pacman/local/python2-numpy-1.5.1-2
Error reading package /var/lib/pacman/local/texlive-bin-2010.1-8
Error reading package /var/lib/pacman/local/lvm2-2.02.85-2
Error reading package /var/lib/pacman/local/sed-4.2.1-3
Error reading package /var/lib/pacman/local/dash-0.5.6.1-2
Error reading package /var/lib/pacman/local/iana-etc-2.30-1
Error reading package /var/lib/pacman/local/acl-2.2.51-1
Error reading package /var/lib/pacman/local/xfprint-4.6.1-5
Error reading package /var/lib/pacman/local/ghostscript-9.02-2
Error reading package /var/lib/pacman/local/expat-2.0.1-6
Error reading package /var/lib/pacman/local/iputils-20101006-1
Error reading package /var/lib/pacman/local/texlive-humanities-2010.22054-1
Error reading package /var/lib/pacman/local/librpcsecgss-0.19-5
Error reading package /var/lib/pacman/local/zziplib-0.13.60-1
Error reading package /var/lib/pacman/local/openmpi-1.5.3-3
Error reading package /var/lib/pacman/local/texlive-formatsextra-2010.22089-1
Error reading package /var/lib/pacman/local/pcmciautils-017-2
Error reading package /var/lib/pacman/local/texlive-publishers-2010.22075-1
Error reading package /var/lib/pacman/local/libglade-2.6.4-2
Error reading package /var/lib/pacman/local/eventlog-0.2.12-2
Error reading package /var/lib/pacman/local/texlive-bibtexextra-2010.22146-1
Error reading package /var/lib/pacman/local/texlive-games-2010.20619-2
Error reading package /var/lib/pacman/local/texlive-pstricks-2010.22140-1
Error reading package /var/lib/pacman/local/subversion-1.6.17-2
Error reading package /var/lib/pacman/local/sysfsutils-2.1.0-6
Error reading package /var/lib/pacman/local/findutils-4.4.2-3
Error reading package /var/lib/pacman/local/rfkill-0.4-2
Error reading package /var/lib/pacman/local/python2-2.7.2-1
Error reading package /var/lib/pacman/local/curl-7.21.6-2
Error reading package /var/lib/pacman/local/musicbrainz-2.1.5-4
Placing 375 nodes
Saving SVG
Rendering PNG
Indexing PNG

I have randomly checked a few of the packages that throw an error, all have special chars in them. :(

**Comment by keenerd** [comment]
Sorry, I have no idea what is causing this and I have never heard of it happening to anyone else.

Since I can't make it happen locally, you are on your own.  Good luck.

**Comment by Blind55** [comment]
Ok, so it appears that $LANG is not the locale variable responsible. Turns out I had only $LANG set to the UTF chars, the others were all "C" (stupid me). Anyways, if you set LC_ALL to a UTF locale, pacgraph works just fine.
TLDR: my fault, pacgraph works as expected with a UTF locale.
